durable-storage/src/persistence_layer.rs

```diff
@@ -155,7 +158,9 @@ impl<'a> PersistenceLayer<'a> {
 
         // To avoid accidentally overwriting an existing database, `error_if_exists` is set.
         let options = rocksdb_creation_options();
-        let db = rocksdb::DB::open(&options, &new_db_path)?;
+        let mut db = rocksdb::DB::open(&options, &new_db_path)?;
+        db.create_cf(BLOB_CF, &rocksdb::Options::default())
+            .expect("A new rocksdb should always be able to create the data column family");
```

Comment:
This is in a `Result` context, might as well propagate the error.

```suggestion
        db.create_cf(BLOB_CF, &rocksdb::Options::default())?;
```